samples/taglist.cpp

```diff
@@ -118,20 +133,37 @@ int main(int argc, char* argv[])
 
             default:
                 rc = 1;
-                break;
+            break;
         }
 
-        if (rc) {
-            std::cout << "Usage: " << argv[0]
-                      << " [--group "
-                         "name|Groups|Exif|Canon|CanonCs|CanonSi|CanonCf|Fujifilm|Minolta|Nikon1|Nikon2|Nikon3|Olympus|"
-                         "Panasonic|Pentax|Sigma|Sony|Iptc"
-                      << "|dc|xmp|xmpRights|xmpMM|xmpBJ|xmpTPg|xmpDM|pdf|photoshop|crs|tiff|exif|aux|iptc|all|ALL]\n"
-                      << "Print Exif tags, MakerNote tags, or Iptc datasets\n";
+        if (rc || bHelp) {
+            out << "Usage: taglist "                                   << std::endl
+                << "           [--group name|"                         << std::endl
+                << "            Groups|Exif|Canon|CanonCs|CanonSi|CanonCf|Fujifilm|Minolta|Nikon1|Nikon2|Nikon3|Olympus|" << std::endl
+                << "            Panasonic|Pentax|Sigma|Sony|Iptc|"                                                        << std::endl
+                << "            dc|xmp|xmpRights|xmpMM|xmpBJ|xmpTPg|xmpDM|pdf|photoshop|crs|tiff|exif|aux|iptc|all|ALL]"  << std::endl
+                << "           [--help] [--checksum]"                  << std::endl
+                << "Print Exif tags, MakerNote tags, or Iptc datasets" << std::endl
+                ;
         }
-        return rc;
     } catch (AnyError& e) {
-        std::cout << "Caught Exiv2 exception '" << e << "'\n";
-        return 1;
+        out << "Caught Exiv2 exception '" << e << "'\n";
+        rc = 1 ;
+    }
+    if ( bChecksum ) {
+#ifdef   EXV_HAVE_LIBZ
+        uLong checksum=0;
+        checksum = adler32(checksum,(const Bytef *)out.str().c_str(),out.str().size());
```

Comment:
If this is only present for testing, then I'd suggest to move this part into the python test suite instead (especially as Python has builtin support for this: https://docs.python.org/3/library/zlib.html#zlib.adler32).

Reply:
You're right.  Delegating this to the test harness is the correct thing to do.